src/archivematicaCommon/lib/databaseFunctions.py

```diff
@@ -229,8 +229,12 @@ def logTaskCompletedSQL(task):
     print("Logging task output to db", task.UUID)
     taskUUID = task.UUID.__str__()
     exitCode = task.results["exitCode"].__str__()
-    stdOut = task.results["stdOut"]
-    stdError = task.results["stdError"]
+    # truncate task stdout/stderror logged to the database
+    # (to prevent mysql server errors, refs. #10732) 
+    TASK_STDOUT_LIMIT = 16*1024*1024
+    TASK_STDERR_LIMIT = 16*1024*1024
+    stdOut = (task.results["stdOut"][:TASK_STDOUT_LIMIT] + '..') if len(task.results["stdOut"]) > TASK_STDOUT_LIMIT else task.results["stdOut"]
+    stdError = (task.results["stdError"][:TASK_STDERR_LIMIT] + '..') if len(task.results["stdError"]) > TASK_STDERR_LIMIT else task.results["stdError"]
```

Comment:
As @sevein points out, if you truncate a Python Unicode string at index 16*1024*1024, you're not guaranteed to get something that, when encoded (e.g., as UTF-8), is 16*1024*1024 bytes. But I don't think you have a hard requirement (e.g., a database constraint) about that number, right @hakamine? You just want to shorten these strings to some fixed approximate maximum. 

    >>> a = 'Jesu\u0301s'
    >>> print(a)
    Jesús
    >>> a.encode('utf8')
    b'Jesu\xcc\x81s'
    >>> len(a)
    6
    >>> a.encode('utf8')
    b'Jesu\xcc\x81s'
    >>> len(a.encode('utf8'))
    7
    >>> a.encode('utf8')[:5]
    b'Jesu\xcc'
    >>> a.encode('utf8')[:5].decode('utf8')
    Traceback (most recent call last):
      File "<stdin>", line 1, in <module>
    UnicodeDecodeError: 'utf-8' codec can't decode byte 0xcc in position 4: unexpected end of data